src/Tracing/Instrumentation/EventSubscriber/AddUserEventSubscriber.php

```diff
@@ -49,18 +49,26 @@ public function onRequestEvent(RequestEvent $event): void
         if ($token && $this->isTokenAuthenticated($token)) {
             $span = $this->mainSpanContext->getMainSpan();
             $user = $token->getUser();
-
             if ($user) {
                 $span->setAttribute(TraceAttributes::ENDUSER_ID, $this->getUsername($user));
-                $span->setAttribute(TraceAttributes::ENDUSER_ROLE, $user->getRoles());
+                $span->setAttribute(TraceAttributes::ENDUSER_ROLE, $this->getRoles($user));
             }
         }
     }
 
     /**
-     * @param UserInterface|object|string $user
+     * @return string[]
      */
-    private function getUsername($user): ?string
+    private function getRoles(UserInterface|string $user): array
```

Comment:
To properly handle [Symfony 5](https://github.com/symfony/symfony/blob/5.0/src/Symfony/Component/Security/Core/Authentication/Token/TokenInterface.php#L54):
```suggestion
    private function getRoles(UserInterface|\Stringable|string $user): array
```